lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -174,7 +174,7 @@ export default function FilePickerApp({ onSubmit }) {
     if (
       shouldSubmit &&
       // We either are not using the deepLinkingAPI, or if we are, wait for deepLinkingFields to be available
-      (!deepLinkingAPI || (deepLinkingAPI && deepLinkingFields))
+      (!deepLinkingAPI || deepLinkingFields)
```

Comment:
If the right hand side of `||` is evaluated, then `deepLinkingAPI` must be truthy.